controller-api/src/main/proto/responsive/controller/internal/v1/controller.proto

```diff
@@ -31,7 +31,9 @@ message PolicyState {
 }
 
 message Action {
-  ChangeApplicationState change_application_state = 1;
+  oneof action {
+    ChangeApplicationState change_application_state = 1;
```

Comment:
do we need a particular upgrade path here? I'm not sure if we've enabled the autoscaler yet for the staging deployment, but if so we should make sure to turn it off and then roll this and the controller

generally we should consider how to make these kind of schema changes once we have customers in prod

Reply:
This is actually a backwards-compatible change (moving a singular field into a `oneof`). But yeah in general I agree.